api/v1alpha2/linodefirewall_types.go

```diff
@@ -70,6 +70,9 @@ type FirewallRule struct {
 	// +kubebuilder:validation:Enum=TCP;UDP;ICMP;IPENCAP
 	Protocol  linodego.NetworkProtocol `json:"protocol"`
 	Addresses *NetworkAddresses        `json:"addresses"`
+	// AddressSetRefs is a list of references to AddressSets
+	// If Addresses is present, AddressSetRefs will be ignored
+	AddressSetRefs []*corev1.ObjectReference `json:"addressSetRefs,omitempty"`
```

Comment:
do these have to be mutually exclusive or could we do both?

Reply:
We could do both if it's less confusing